receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -404,7 +446,28 @@ func (mf *metricFamily) addSeries(seriesRef uint64, metricName string, ls labels
 			mg.createdSeconds = v
 		default:
 			boundary, err := getBoundary(mf.mtype, ls)
-			if err != nil {
+			if err == errEmptyLeLabel {
+				// Check if this is a VictoriaMetrics histogram.
+				// We only check this if the "le" label was missing so we don't pay for the VM checks in the classic prom cases.
+				vmrange := ls.Get(vmHistogramRangeLabel)
+				if vmrange == "" {
+					// This is not a VM histogram. Return the original error about "le" being missing.
+					return errEmptyLeLabel
+				}
+				start, end, err := vmHistogramParseRange(vmrange)
+				if err != nil {
+					return err
+				}
+				// Use the ratio of end/start to infer the scale factor we should use for the resulting ExponentialHistogram.
+				// Ignore the "zero" bucket since we can't infer the scale from it.
+				// We assume that the scale is the same for all other (non-zero) buckets.
+				// If for some reason they're different, the best we can do is pick one of them.
+				if start != 0 && end != 0 {
+					mg.vmHistScale = vmHistogramGetScale(start, end)
```

Comment:
is it worth logging this case, if we get different scales from the same vm_histo series?